lib/ce/src/hot-settings-resolver.service.ts

```diff
@@ -47,44 +47,30 @@ const AVAILABLE_HOOKS: string[] = ['afterBeginEditing', 'afterCellMetaReset', 'a
 
 @Injectable()
 export class HotSettingsResolver {
-  private hooks = AVAILABLE_HOOKS;
-  private options = AVAILABLE_OPTIONS;
 
   mergeSettings(component): object {
     const mergedSettings: object = {};
+    const options = AVAILABLE_HOOKS.concat(AVAILABLE_OPTIONS);
 
-    if (component['settings'] !== void 0) {
-      Object.keys(component['settings']).forEach((key) => {
-        if (this.hooks.indexOf(key) > -1) {
-          mergedSettings[key] = (p1, p2, p3, p4, p5, p6) => {
-            return component._ngZone.run(() => {
-              return component['settings'][key].call(component.hotInstance, p1, p2, p3, p4, p5, p6);
-            })
-          };
+    options.forEach((key) => {
+      let option = component['settings'] && component['settings'][key];
```

Comment:
This expression can be risky in cases where `component['settings']` contains different value than an object. Could you secure this by adding `typeof` or something similar to the equalization?